clients/src/main/java/org/apache/kafka/common/requests/TxnOffsetCommitRequest.java

```diff
@@ -45,9 +45,39 @@
 
         public final TxnOffsetCommitRequestData data;
 
-        public Builder(TxnOffsetCommitRequestData data) {
+        public Builder(final String transactionalId,
+                       final String consumerGroupId,
+                       final long producerId,
+                       final short producerEpoch,
+                       final Map<TopicPartition, CommittedOffset> pendingTxnOffsetCommits) {
+            this(transactionalId,
+                consumerGroupId,
+                producerId,
+                producerEpoch,
+                pendingTxnOffsetCommits,
+                JoinGroupRequest.UNKNOWN_MEMBER_ID,
```

Comment:
See my other comment: it seems we initialize generationId / memberId as UNKNOWN anyways, which means that if enableFencing is false it would stay as `UNKNOWN_XX` and the broker would not check its generationId or memberId, so it seem we can get rid of the boolean flag indeed since if only groupId is passed in, the other fields' default value is sufficient to bypass the fencing.

Reply:
Yep, we could just get rid of the check entirely, as for groupId based commit the other fields will all be default.